### PR TITLE
doc: Improve Javadoc quality in Environment interface

### PR DESCRIPTION
**Pull request by slarse**
#3923 

The `Environment` interface is arguably one of the most important ones in the entire Spoon library, and it's fairly underdocumented. This PR improves the documentation by adding all missing `@param` and `@return` tags, fixing errors and adding missing explanations of what things actually do.

Javadoc method quality score increases like so:

```
JAVADOC QUALITY SCORE (lower is better)
    Compare: 2641
    Current: 2584
```

There are still improvements to be made here, especially in explaining what the options you can set actually do, but I felt like this was enough for a single PR.

### CODE REVIEW

File: src/main/java/spoon/compiler/Environment.java [hunk 1]
```diff
@@ -142,60 +170,83 @@
...
 	 * Sets the size of the tabulations in the generated source code.
+	 *
+	 * @param size tabulation size to set
 	 */
 	void setTabulationSize(int size);
 
 	/**
 	 * Tells if Spoon uses tabulations in the source code.
+	 *
+	 * @return true iff Spoon uses tabsulationswhen pretty-printing
```
Comment (andrewbwogi) [inline src/main/java/spoon/compiler/Environment.java:218]: Hi @slarse, you have a typo here: `tabsulationswhen`

Comment (andrewbwogi) [inline src/main/java/spoon/compiler/Environment.java:218]:
```suggestion
	 * @return true iff Spoon uses tabulations when pretty-printing
```

Comment (slarse) [reply src/main/java/spoon/compiler/Environment.java]: Thanks!

**Comment by monperrus** [comment]
We :heart: documentation PR. Thanks Simon.

**Pull request merged**